### PR TITLE
[1LP][WIP]adding wait_for_elements into paginator pages - fix for 5.9

### PR DESCRIPTION
**Pull request by lkhomenk**
Purpose or Intent
=================

__Fixing__ page surfing with paginator because it is runs too fast - we get empty entities_names because UI elements haven't been loaded
This affects 5.9 only
no testing at the moment except :

```
view = navigate_to(Vm, 'All')
view.toolbar.view_selector.selected
Out[10]: u'Tile View'
view.entities.paginator.items_amount (at the moment this shows - how many items are shown on page, not the whole amount)
Out[11]: 20
view.entities.paginator.pages_amount
Out[12]: 56
view.entities.paginator.set_items_per_page(100)
view.entities.paginator.items_amount <<<<<<<<<< this is the issue. this line was called while UI was loading page with 100 items on it
Out[14]: 0
view.entities.paginator.items_amount
Out[15]: 100
view.entities.get_entity(by_name='webmks-photonos-kkulkarn',surf_pages=True)
Out[16]: <widgetastic_manageiq.JSBaseEntity at 0x7f92069ecb10>
view.toolbar.view_selector.select('Tile View')
view.entities.paginator.pages_amount
Out[19]: 12
view.entities.get_entity(by_name='webmks-photonos-kkulkarn',surf_pages=True)
Out[20]: <widgetastic_manageiq.JSBaseEntity at 0x7f91ebcd65d0>
view.toolbar.view_selector.select('List View')
view.entities.paginator.items_per_page
Out[22]: 100
view.entities.get_entity(by_name='webmks-photonos-kkulkarn',surf_pages=True)
Out[23]: <widgetastic_manageiq.JSBaseEntity at 0x7f91ebccd410>
```

### CODE REVIEW

**Comment by lkhomenk** [comment]
5.7/5.8 are not affected as changed apply to JS paginator

File: widgetastic_manageiq/__init__.py [hunk 1]
```diff
@@ -1522,6 +1523,12 @@ def pages(self):
 
             # Adding 1 to pages_amount to include the last page in loop
             for page in range(1, self.pages_amount + 1):
+                # JSPaginatinoPane will wait for elements to load after changing page
+                # miq-tile-selection covers both - Grid and Tile view
+                if sel.is_displayed('//*[@class="miq-data-table"]'):
```
Comment (lkhomenk) [inline widgetastic_manageiq/__init__.py]: maybe it could be more elegant. and container holding 'entities' could be stored. ideas?

**Comment by lkhomenk** [comment]
@mfalesni this PR could be just a workaround. and some wait could  be added into widgets.

File: widgetastic_manageiq/__init__.py (hunk 1 above)
Comment (izapolsk) [reply widgetastic_manageiq/__init__.py]: @lkhomenk, have you tried ensure_page_safe()?

btw, whether Karel is aware of this issue ? I haven't run into it when tested JS api.

Comment (lkhomenk) [reply widgetastic_manageiq/__init__.py]: @izapolsk was pytest_selemiun counted as web_ui? @mfalesni  talked to sme DEVs, solution - add mechanism to check element is updated

Comment (izapolsk) [reply widgetastic_manageiq/__init__.py]: @lkhomenk, sure, this is part of it

Comment (lkhomenk) [reply widgetastic_manageiq/__init__.py]: browser.plugin.ensure_page_safe doesn't help at all. paginator goes through all pages and entities can't have a chance to load. @izapolsk

Comment (mfalesni) [reply widgetastic_manageiq/__init__.py]: @lkhomenk pytest_selenium counts as web_ui and is definitely unwanted ... use `self.browser` instead.

Comment (mfalesni) [reply widgetastic_manageiq/__init__.py]: @izapolsk From what I have talked with Martin H. we will have less and less opportunities for JS checking as they move towards components.

**Comment by izapolsk** [comment]
@lkhomenk, thank you for devoting time to that issue.
honestly, I don't like such solution and I guess there should be something better.

**Comment by izapolsk** [comment]
@lkhomenk, Karel and me are trying to find better way to fix this issue.

**Comment by mshriver** [comment]
Moving to 1LP+WIP to block PRT.  @izapolsk sounds like he's rejecting the PR, so I'll leave WIP and not review unless this becomes the preferred solution.

**Comment by izapolsk** [comment]
@mshriver, we already discussed this issue and fix with @lkhomenk offline and I already provided with devs help a bit better fix #5706.

**Pull request closed without merging**